feat: #2 프로젝트 초기 설정

tpleehan opened the pull request:
## #2 프로젝트 초기 설정
- [x] main branch → develop (default)
- [x] Naver Hackday Convetion 적용(IDE: IntelliJ 적용 완료)
- [x] Swagger 적용 (Gradle dependency 추가)

### Coding Convention
코드의 가독성을 높이기 위해선
자신만의 익숙한 코드 스타일이 아닌 다른 사람이 보더라도 이해하기 쉬운 코드가 가독성이 높은 코드이다.
일관성 있는 규칙을 위해 컨벤션을 진행하는데 다양한 코딩 컨벤션이 있다.
그중 _Naver Hackday Convention_ 으로 설정한다.

### Restful API 문서화 도구
API 문서는 협업하고 있는 개발자는 물론 (예: 프론트엔드 개발자 등)
사용자에게 제공한 API를 파악하기 쉽게 문서화를 한다.
메모장, 엑셀, 한글, 워드, 노션 등 여러 문서 편집기를 통해 전달할 수 있지만
문서를 작성하는 데 있어 누락되는 경우는 물론 가독성이 떨어지는 경우가 있다.
API 스펙에 대한 문서를 자동화 지원이 되는 OAS(Open Api Specification)를 위한 프레임워크인
_Swagger_ 를 설정하여 프로젝트를 진행한다.

tpleehan committed:
feat: #2 Project init setting

프로젝트 초기 설정
- Coding Convetion / naver hackday conventions 설정
- Restful API 문서화 도구 / Swagger 설정

brody-0125 commented on build.gradle:
```diff
@@ -1,26 +1,28 @@
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.springframework.session:spring-session-core'
+    implementation group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.9.2'
+    implementation group: 'io.springfox', name: 'springfox-swagger2', version: '2.9.2'
```
springfox를 사용하신 이유가 있으실까요? springfox는 20년도 이후로 업데이트가 되지 않고 있는데요.
최근에도 업데이트 되고 있는 https://mvnrepository.com/artifact/org.springdoc/springdoc-openapi-ui 를 사용하시는건 어떨까요?

brody-0125 replied:
접근 경로를 설정해서 문서가 정상적으로 뜨는지 확인하는 것도 좋을 것 같습니다. ㅎㅎ

tpleehan replied:
MVN Repository에서 Swagger로 검색 했을 때 SpringFox가 노출 되기도 했고
블로그 글을 참고해서 적용 하다 보니 예전 버전으로 적용을 해버렸네요. ㅠㅠ

## Swagger 설정
Swagger는 `Spring-Fox`, `Spring-Doc` 2가지 라이브러리가 있다.
- Spring-Fox
  - 2015년에 출시된 라이브러리이다.
  - 2020년 7월 이후로 업데이트가 멈춘 상태이다.
  - 또한, Spring Boot 2.6이상 버전에서는 오류로 인한 이슈가 있다.
- Spring-Doc
  - 2019년에 출시된 라이브러리이다.
  - 최근까지 업데이트가 되고 있다. (2022년 10월)
  - Spring Boot 2.6이상 버전에서 잘 작동한다.

### 참고 사항
[Springfox 3.0.0 is not working with Spring Boot 2.6.0](https://stackoverflow.com/questions/70178343/springfox-3-0-0-is-not-working-with-spring-boot-2-6-0)

brody-0125 replied:
해당 부분 라이브러리 변경 후에 푸시해주세요~

tpleehan committed:
build: #2 Project init setting - Swagger 및 gitignore 추가

프로젝트 초기 설정
- Swagger / springfox -> springdoc 변경(최신 버전 적용)
- application.properties 설정 파일 gitignore 추가

tpleehan committed:
feat: #2 Project init setting - Swagger 및 gitignore 추가

프로젝트 초기 설정
- Swagger / springfox -> springdoc 변경(최신 버전 적용)
- application.properties 설정 파일 gitignore 추가

tpleehan committed:
Merge remote-tracking branch 'origin/feature/2-project-init-setting' into feature/2-project-init-setting
